website/content/en/highlights/2022-03-15-vrl-vm-beta.md

```diff
@@ -0,0 +1,122 @@
+---
+date: "2022-03-15"
+title: "Beta release of a Virtual Machine to run VRL programs"
+description: "A faster way to run VRL"
+authors: ["StephenWakely"]
+pr_numbers: [9829]
+release: "0.21.0"
+hide_on_release_notes: false
+badges:
+  type: enhancement
+---
+
+As part of our performance improvements in Vector, we have developed a new Virtual Machine for
+running VRL programs.
+
+In our soak tests we are getting performance improvements of between 10-15% depending on
```

Comment:
Yeah, it's probably worth noting that VRL itself is sped up up to 2x.

You could also mention that this will be the new default (thus encouraging more people to try it out).

Reply:
That was the potential improvements. After adding a lot of the safety measures to the VM the performance improvement did reduce a fair bit. For example we had to add a check after each statement to check if an error had occurred and if so jump out of the block. There's a fair amount of scope for recouping a lot of this by making the compiler more intelligent, but that is work that still needs doing.